lib/utils/storage_directory_utils.dart

```diff
@@ -0,0 +1,21 @@
+import 'package:path_provider/path_provider.dart';
+
+class StorageDirectoryUtils {
+  StorageDirectoryUtils._();
+
+  static final StorageDirectoryUtils _instance = StorageDirectoryUtils._();
+
+  static StorageDirectoryUtils get instance => _instance;
+
+  Future<String> getFileStoreDirectory() async {
+    try {
+      try {
+        return (await getTemporaryDirectory()).path;
+      } catch (_) {
+        return (await getApplicationDocumentsDirectory()).path;
+      }
+    } catch (_) {
+      return (await getDownloadsDirectory())!.path;
```

Comment:
how can this be returned ? the inner try catch wont throw any exception if i'm correct

Reply:
it can. because some platforms may don't support temporary directory or document directory. you can reference here https://pub.dev/packages/path_provider